stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/composer/MessageComposerView.kt

```diff
@@ -0,0 +1,195 @@
+package io.getstream.chat.android.ui.message.composer
+
+import android.content.Context
+import android.text.Editable
+import android.util.AttributeSet
+import android.view.View
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.core.view.isVisible
+import com.google.android.material.internal.TextWatcherAdapter
+import io.getstream.chat.android.client.models.Attachment
+import io.getstream.chat.android.common.state.MessageAction
+import io.getstream.chat.android.core.ExperimentalStreamChatApi
+import io.getstream.chat.android.ui.common.extensions.internal.streamThemeInflater
+import io.getstream.chat.android.ui.databinding.StreamUiMessageComposerBinding
+import io.getstream.chat.android.ui.databinding.StreamUiMessageComposerDefaultCenterContentBinding
+import io.getstream.chat.android.ui.databinding.StreamUiMessageComposerDefaultLeadingContentBinding
+import io.getstream.chat.android.ui.databinding.StreamUiMessageComposerDefaultTrailingContentBinding
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+
+@ExperimentalStreamChatApi
+public class MessageComposerView : ConstraintLayout {
+
+    private lateinit var binding: StreamUiMessageComposerBinding
+
+    /**
+     * @property messageInputState The current state of the [MessageComposerView] distributed using [Flow] API.
+     *
+     * Whenever the value of this property changes updated the [MessageComposerView] re-renders automatically.
+     * If you are using custom [View] implementation for [leadingContent], [centerContent], or [trailingContent] you may
+     * want to collect [messageInputState] items to update them when the state changes.
+     */
+    public val messageInputState: MutableStateFlow<MessageInputState> = MutableStateFlow(MessageInputState())
+
+    /**
+     * @property onSendMessageAction Handler invoked when the user taps on the send message button.
+     */
+    public var onSendMessageAction: () -> Unit = {}
+
+    /**
+     * @property onCancelAction Handler invoked when the user cancels the active action (in Reply or Edit modes).
+     */
+    public var onCancelAction: () -> Unit = {}
+
+    /**
+     * @property onInputTextChanged Handler invoked when the user enters text in message input field.
+     */
+    public var onInputTextChanged: (String) -> Unit = {}
+
+    private val defaultLeadingContent: View by lazy {
```

Comment:
Do we need this default leading content as a property or can it be somehow lazily added to the UI when the user starts setting things up?

It seems it's not "lazy" at all, given that we immediately create the `leadingContent` within the `init()` block and create this property. Maybe it can all be inlined there?

Reply:
`lazy` will make sense only when user will be able to decide to hide one of content views using style or attrs-based API. Style will be parsed in init() too. Otherwise we are not able to make it effectively lazy.